src/components/component-playground.jsx

```diff
@@ -351,7 +358,10 @@ module.exports = React.createClass({
 
   componentWillReceiveProps: function(nextProps) {
     if (this.constructor.didFixtureChange(this.props, nextProps)) {
-      this.setState(this.constructor.getFixtureState(nextProps));
+      var nextState = this.constructor.getFixtureState(nextProps);
+      this.setState(nextState);
+      store.dispatch(resetStore(this.props.reduxStore));
+      store.dispatch(changeFixture(nextState.fixtureContents));
```

Comment:
Actions can't be dispatched from the reducer, so a `resetStore` action cannot be dispatched when resolving `changeFixture`.

Also, separate actions seem to give a better idea on how the store changes.